### DISCUSSION TITLE
What is Match?

### DISCUSSION THREAD

**Discussion by jgarte** [qtile/qtile#4134] [Q&A]
The docs say:

> Match for dynamic groups or auto-floating windows.

Is there other explanations of what [Match](https://docs.qtile.org/en/stable/_modules/libqtile/config.html#Match) is that give more introduction to the concept of `Match` in a more discoursed way?

I can imagine what `Match` might be but the above description does not define it. It presents the concept like the reader already knows what `Match` is and also does not exemplify ways that it can be used by the user of the qtile API.

**Comment by jrwrigh** [comment]
I think the best way to put it is that a `Match` A) contains a description of some potential Window and B) provides a method `Match.compare` to determine whether a given window matches.

I agree that the first line of the doc isn't very good and should probably be changed.

**Reply by jgarte** [reply]
@jrwrigh Thanks for the reply. It's much appreciated!

What does it mean to match? Why would we want to match windows at all? I think that's the concept that I'm trying to spell out that isn't explicitly defined anywhere. For example, imagine you're a GNOME or Microsoft Windows user that is going to use qtile for the first time and doesn't know anything about the concept of `Match` in qtile. A definition of explaining the concept of `Match` in the context of using qtile would make it easier for them to understand what it is.

That said, I've used dwm, i3, stumpwm and other window managers casually over the years and I'm having trouble understanding what qtile means by `Match` because the concept is not explicitly defined anywhere... ;()

**Reply by jrwrigh** [reply]
I've created a PR that should clarify the documentation a bit: #4144

**Reply by jgarte** [reply]
Thanks for that!

> to determine if  its properties *match*. It can match by title, wm_class, role, wm_type,   wm_instance_class or net_wm_pid.

If they match by a property what are some example things that you are able to do after the match? That's the key thing I'm trying to understand ;()

I understand what it means to match with something but once I match what are some examples of things I can I do, trigger, hook, etc?

And if I'm able to trigger, hook, etc something then what are example things that I can trigger of hook?
